### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by pgrimaud**
Hi,

This huge PR will fix : 

- 1 typo in `.eslintrc.js`
- 1 typo in `docs/docs/deploying-to-now.md`
- 1 typo in `docs/tutorial/e-commerce-with-datocms-and-snipcart/index.md`
- 1 typo in `examples/using-i18n/gatsby-node.js`
- 1 typo in `examples/using-js-search/src/components/SearchContainer.js`
- 1 typo in `packages/gatsby-dev-cli/src/utils/traverse-package-deps.js`
- 1 typo in `packages/gatsby-plugin-catch-links/src/__tests__/catch-links.js`
- 1 typo in `packages/gatsby-recipes/README.md`
- 1 typo in `packages/gatsby/src/query/query-watcher.js`
- 1 typo in `packages/gatsby/src/schema/extensions/index.js`
- 1 typo in `packages/gatsby/src/utils/__tests__/api-runner-error-parser.ts`
- 1 typo in `packages/gatsby/src/utils/webpack.config.js`

Ciao! :robot:

### CODE REVIEW

**Comment by pgrimaud** [comment]
Ref to https://github.com/gatsbyjs/gatsby-i18n-source/pull/8 and more!

**Comment by AishaBlake** [comment]
Hmm, it doesn't seem like this change should be causing issues but I'm getting this error: `Error: Type with name "CreatorsYaml" does not exists`. I'm going to leave this to the team for a more thorough review!

File: packages/gatsby/src/query/query-watcher.js [hunk 1]
```diff
@@ -165,7 +165,7 @@ const clearInactiveComponents = () => {
 
   const activeTemplates = new Set()
   pages.forEach(page => {
-    // Set will guarantee uniqeness of entires
+    // Set will guarantee uniqeness of entries
```
Comment (ascorbic) [inline packages/gatsby/src/query/query-watcher.js:168]:
```suggestion
    // Set will guarantee uniqueness of entries
```

**Comment by muescha** [comment]
@pgrimaud there is one merge conflict now:

- docs/docs/deploying-to-now.md

**Comment by pgrimaud** [comment]
@muescha I fixed conflict by rebasing from upstream. 👍

**Comment by wardpeet** [comment]
@pgrimaud seems like tests are failing with your changes, could you update them?

**Comment by muescha** [comment]
all test logs now (since some days) require a login and are not visible

**Comment by pgrimaud** [comment]
@wardpeet I fixed tests 👍

**Pull request merged**